issue_processing/jira_issue_processing.py

```diff
@@ -70,52 +70,82 @@ def run():
-    # 1) load the list of issues
-    issues = load_xml(__srcdir)
-    # 1b) load the list of persons
+    # load the list of persons
     persons = load_csv(__srcdir_csv)
-    # 2) re-format the issues
-    issues = parse_xml(issues, persons, args.skip_history)
-    # 3) load issue information via api
-    if not args.skip_history:
-        load_issue_via_api(issues, persons, __conf["issueTrackerURL"])
-    # 4) update user data with Codeface database
-    # mabye not nessecary
-    issues = insert_user_data(issues, __conf)
-    # 5) dump result to disk
-    print_to_disk(issues, __resdir)
-    # 6) export for Gephi
-    print_to_disk_gephi(issues, __resdir)
-    # 7) export for jira issue extraction to use them in dev-network-growth
-    print_to_disk_extr(issues, __resdir)
-    # 8) dump bug issues to disk
-    print_to_disk_bugs(issues, __resdir)
+
+    # load the xml-file list
+    file_list = [f for f in os.listdir(__srcdir) if os.path.isfile(os.path.join(__srcdir, f))]
+
+    # creates empty result files
+    clear_result_files(__resdir)
+
+    # processes every xml-file
+    for file in file_list:
+        # 1) load the list of issues
+        issues = load_xml(__srcdir, file)
+        # 2) re-format the issues
+        issues = parse_xml(issues, persons, args.skip_history)
+        # 3) load issue information via api
+        if not args.skip_history:
+            load_issue_via_api(issues, persons, __conf["issueTrackerURL"])
+        # 4) update user data with Codeface database
+        # mabye not nessecary
+        issues = insert_user_data(issues, __conf)
+        # 5) dump result to disk
+        print_to_disk(issues, __resdir)
+        # 6) export for Gephi
+        print_to_disk_gephi(issues, __resdir)
+        # 7) export for jira issue extraction to use them in dev-network-growth
+        print_to_disk_extr(issues, __resdir)
+        # 8) dump bug issues to disk
+        print_to_disk_bugs(issues, __resdir)
 
     log.info("Jira issue processing complete!")
 
 
-def load_xml(source_folder):
+def clear_result_files(results_folder):
+    """
+    Creates an empty csv file for every result file.
+    :param results_folder: the folder where to save the result files
+    """
+
+    log.info("Clear result files ...")
+
+    # construct path to output files
+    output_file = os.path.join(results_folder, "issues-jira.list")
+    output_file_bugs = os.path.join(results_folder, "bugs-jira.list")
+    output_file_extr = os.path.join(results_folder, "issue-jiras.list")
+    output_file_gephi_edges = os.path.join(results_folder, "issues-jira-gephi-edges.csv")
+    output_file_gephi_nodes = os.path.join(results_folder, "issues-jira-gephi-nodes.csv")
+
+    # creates empty csv files
+    csv_writer.write_to_csv(output_file, [])
+    csv_writer.write_to_csv(output_file_bugs, [])
+    csv_writer.write_to_csv(output_file_extr, [])
+    csv_writer.write_to_csv(output_file_gephi_edges, [])
+    csv_writer.write_to_csv(output_file_gephi_nodes, [])
```

Comment:
Is there a reason why you do not use a list for the files and for-loop for the file creation/truncation here?

For example, `open(result_file, "w+").close()` would either create a non-existent file `result_file` or truncate it if it exists already (i.e., remove all content). This would make a loop-version very concise and short.